tests/config/test_DeployConfig.py

```diff
@@ -0,0 +1,261 @@
+from __future__ import annotations
+
+from pathlib import Path
+from unittest import mock
+from unittest.mock import mock_open
+
+import pytest
+
+from schemachange.config.BaseConfig import BaseConfig
+from schemachange.config.DeployConfig import DeployConfig
+
+
+@mock.patch("pathlib.Path.is_dir", side_effect=[False])
+def test_invalid_root_folder(_):
+    with pytest.raises(Exception) as e_info:
+        DeployConfig.factory(
+            config_file_path=Path("some_config_file_name"),
+            root_folder="some_root_folder_name",
+            modules_folder="some_modules_folder_name",
+            config_vars={"some": "config_vars"},
+            snowflake_account="some_snowflake_account",
+            snowflake_user="some_snowflake_user",
+            snowflake_role="some_snowflake_role",
+            snowflake_warehouse="some_snowflake_warehouse",
+            snowflake_database="some_snowflake_database",
+            snowflake_schema="some_snowflake_schema",
+            change_history_table="some_history_table",
+            query_tag="some_query_tag",
+            oauth_config={"some": "values"},
+        )
```

Comment:
I suggest moving these `factory()` calls to [fixtures](https://docs.pytest.org/en/stable/explanation/fixtures.html) to make the tests simpler.

Reply:
I'm not a pytest pro, but I don't think a fixture would work. Fixtures are instantiated prior to the test and I need to assert on an exception in many of these tests. Also, we would need to parameterize the fixtures, pushing the vanilla class instantiation into pytest's parameterization syntax. I find this more difficult to read. Instead, I bundled default arguments to avoid too much repetition.